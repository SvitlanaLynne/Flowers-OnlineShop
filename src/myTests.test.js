/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, fireEvent } from "@testing-library/react";

const MockProducts = ({ data }) => {
  return (
    <div>
      {data.map((flower) => (
        <div key={flower.id}>
          {flower.name}, {flower.category}, {flower.color}
        </div>
      ))}
      <button onClick={() => console.log("Button clicked")}>Click Me</button>
    </div>
  );
};

test("The array of selected filters is filled when the filter is selected", () => {
  const data = [
    {
      id: 1,
      name: "Buttercup",
      color: "yellow",
      category: "wedding",
      description: "A vibrant yellow flower, perfect for weddings.",
      picture: "buttercup.png",
      price: 5.0,
    },
    {
      id: 2,
      name: "Crocus",
      color: "purple",
      category: "spring",
      description: "A lovely purple flower that blooms in spring.",
      picture: "crocus.png",
      price: 6.0,
    },
    {
      id: 3,
      name: "Daffodil",
      color: "yellow",
      category: "spring",
      description: "Bright yellow spring flower.",
      picture: "daffodil.png",
      price: 7.0,
    },
  ];

  const { getByRole, container } = render(
    <div>
      <MockProducts data={data} />
    </div>,
  );
  const button = getByRole("button");
  fireEvent.click(button);

  expect(container.querySelector("button")).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
