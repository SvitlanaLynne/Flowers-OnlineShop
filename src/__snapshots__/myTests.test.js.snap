// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The array of selected filters is filled when the filter is selected 1`] = `
<div>
  <div>
    <div>
      <div>
        Buttercup
        , 
        wedding
        , 
        yellow
      </div>
      <div>
        Crocus
        , 
        spring
        , 
        purple
      </div>
      <div>
        Daffodil
        , 
        spring
        , 
        yellow
      </div>
      <button>
        Click Me
      </button>
    </div>
  </div>
</div>
`;
